### PR TITLE
refactor(ui): add shared components to the ui workspace

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Copy the Header, Footer and StatusBar components to the ui workspace.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Run the react client and check that the header footer and status bar all work.

## Fixes

Fixes: canonical-web-and-design/app-tribe#954.

### CODE REVIEW

File: ui/src/app/base/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  }
+}
+
+type Props = {
+  debug?: boolean;
+  enableAnalytics?: boolean | null;
+  version: string;
+};
+
+export const Footer = ({
```
Comment (huwshimi) [inline ui/src/app/base/components/Footer/Footer.tsx:20]: I've copied these components over and not removed the ones in shared as they're still required by legacy.

File: ui/src/app/base/components/Header/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+        link.inHardwareMenu && !forHardwareMenu,
+    }),
+    isSelected: isSelected(path, link, appendNewBase),
+    key: link.url,
+    label: link.label,
+    url: link.url,
+  }));
+};
+
+export const Header = ({
```
Comment (huwshimi) [inline ui/src/app/base/components/Header/Header.tsx:141]: This component has some minor changes to remove the `rootScope` prop and related `useEffect` handlers.

File: ui/src/app/base/components/Header/Header.tsx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+  const isAuthenticated = !!authUser;
+  // Hide the navigation items when the user is not authenticated or hasn't been
+  // through the intro process.
+  const showLinks = isAuthenticated && completedIntro;
+
+  useEffect(() => {
+    if (!debug && enableAnalytics && uuid && version && authUser) {
+      (function (w, d, s, l, i) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```
Comment (Caleb-Ellis) [inline ui/src/app/base/components/Header/Header.tsx:164]: Surely there's a way to disable linting for a block of code?

**Review by Caleb-Ellis — APPROVED** [review]
LGTM!

File: ui/src/app/base/components/Header/Header.tsx (hunk 3 above)
Comment (huwshimi) [reply ui/src/app/base/components/Header/Header.tsx:164]: Unfortunately not: https://github.com/Microsoft/TypeScript/issues/19573

Comment (Caleb-Ellis) [reply ui/src/app/base/components/Header/Header.tsx:164]: Madness!

**Pull request merged**